includes/Core/Conversion_Tracking/REST_Conversion_Tracking_Controller.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Conversion_Tracking\REST_Conversion_Tracking_Controller
+ *
+ * @package   Google\Site_Kit\Core\Conversion_Tracking
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Conversion_Tracking;
+
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\REST_API\REST_Route;
+use Google\Site_Kit\Core\REST_API\REST_Routes;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * Class for handling rest routes for Conversion Tracking settings.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class REST_Conversion_Tracking_Controller {
+
+	/**
+	 * Conversion_Tracking_Settings instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Conversion_Tracking_Settings
+	 */
+	protected $settings;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Conversion_Tracking_Settings $settings Conversion Tracking settings.
+	 */
+	public function __construct( Conversion_Tracking_Settings $settings ) {
+		$this->settings = $settings;
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_filter(
+			'googlesitekit_rest_routes',
+			function ( $routes ) {
+				return array_merge( $routes, $this->get_rest_routes() );
+			}
+		);
+
+		add_filter(
+			'googlesitekit_apifetch_preload_paths',
+			function ( $paths ) {
+				return array_merge(
+					$paths,
+					array(
+						'/' . REST_Routes::REST_ROOT . '/core/site/data/conversion-tracking',
+					)
+				);
+			}
+		);
+	}
+
+	/**
+	 * Gets REST route instances.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return REST_Route[] List of REST_Route objects.
+	 */
+	protected function get_rest_routes() {
+		$has_capabilities = function() {
+			return current_user_can( Permissions::VIEW_DASHBOARD );
```

Comment:
Why was this capability chosen and applied to both readable and editable methods?

`VIEW_DASHBOARD` means either authenticated or view-only dashboard, which is more open than it should be, at least for updating. Aside from that though, I don't think this data would be used anywhere except module settings. It seems that `MANAGE_OPTIONS` would be more appropriate. If this isn't the case we should add a note as to why.

Reply:
Oh, that's totally true, I just missed this. I've filed #8738.